openmdao/solvers/nonlinear/broyden.py

```diff
@@ -336,24 +367,46 @@ def _iter_get_norm(self):
         Returns
         -------
         float
-            norm.
+            Norm of the residuals.
         """
         # Need to cache the initial residuals, which is done in this function.
         fxm = self.get_residuals()
         if not self._full_inverse:
             # Use full model residual for driving the main loop convergence.
             fxm = self._system._residuals._data
 
-        if self._system.under_complex_step:
-            return (np.sum(fxm.real**2) + np.sum(fxm.imag**2))**0.5
-        else:
-            return np.sum(fxm**2) ** 0.5
+        return self.compute_norm(fxm)
+
+    def compute_norm(self, vec):
+        """
+        Compute norm of the vector.
+
+        Under MPI, compute the norm on rank 0, and broadcast it to all other ranks.
+
+        Parameters
+        ----------
+        vec : ndarray
+            Array of real or complex values. For MPI on rank 0, should be full dimension of the
+            openmdao vector with duplicate indices removed.
+
+        Returns
+        -------
+        float
+            Norm of vec, computed on rank 0 and broadcast to all other ranks.
+        """
+        system = self._system
+
+        if system.comm.size > 1:
+            # Norms computed for all vars on rank 0, then broadcast out.
+            mpi_typ = MPI.C_DOUBLE_COMPLEX if system.under_complex_step else MPI.DOUBLE
+            system.comm.Bcast((vec, mpi_typ), root=0)
```

Comment:
Couldn't we just compute the norm locally here and broadcast that out instead of broadcasting out the whole vector?

Reply:
Fixed.